src/gui/vector/qgsattributesformmodel.h

```diff
@@ -0,0 +1,738 @@
+
+        /**
+         * Returns the HTML editor configuration.
+         *
+         * \see setHtmlElementEditorConfiguration()
+         */
+        HtmlElementEditorConfiguration htmlElementEditorConfiguration() const;
+
+        /**
+         * Sets the HTML editor configuration.
+         *
+         * \see htmlElementEditorConfiguration()
+         */
+        void setHtmlElementEditorConfiguration( const HtmlElementEditorConfiguration &htmlElementEditorConfiguration );
+
+        /**
+         * Returns the spacer element configuration
+         * \since QGIS 3.30
+         */
+        SpacerElementEditorConfiguration spacerElementEditorConfiguration() const;
+
+        /**
+         * Sets the the spacer element configuration to \a spacerElementEditorConfiguration
+         * \since QGIS 3.30
+         */
+        void setSpacerElementEditorConfiguration( SpacerElementEditorConfiguration spacerElementEditorConfiguration );
+
+        /**
+         * Returns the background color of a container.
+         *
+         * \see setBackgroundColor()
+         */
+        QColor backgroundColor() const;
+
+        /**
+         * Sets the background color of a container.
+         *
+         * \see backgroundColor()
+         */
+        void setBackgroundColor( const QColor &backgroundColor );
+
+        /**
+         * Returns the editor configuration for text element.
+         * \since QGIS 3.30
+         */
+        TextElementEditorConfiguration textElementEditorConfiguration() const;
+
+        /**
+         * Sets the editor configuration for text element to \a textElementEditorConfiguration.
+         * \since QGIS 3.30
+         */
+        void setTextElementEditorConfiguration( const TextElementEditorConfiguration &textElementEditorConfiguration );
+
+      private:
+        int mColumnCount = 1;
+        Qgis::AttributeEditorContainerType mContainerType = Qgis::AttributeEditorContainerType::Tab;
+        bool mShowLabel = true;
+        int mHorizontalStretch = 0;
+        int mVerticalStretch = 0;
+        QgsOptionalExpression mVisibilityExpression;
+        RelationEditorConfiguration mRelationEditorConfiguration;
+        QmlElementEditorConfiguration mQmlElementEditorConfiguration;
+        HtmlElementEditorConfiguration mHtmlElementEditorConfiguration;
+        TextElementEditorConfiguration mTextElementEditorConfiguration;
+        SpacerElementEditorConfiguration mSpacerElementEditorConfiguration;
+        QColor mBackgroundColor;
+        bool mCollapsed = false;
+        QgsOptionalExpression mCollapsedExpression;
+        QgsAttributeEditorElement::LabelStyle mLabelStyle;
+    };
+};
+
+
+/**
+ * \brief Class to hold parent-child relations as well as node data contained within a QgsAttributesFormModel.
+ *
+ * \warning Not part of stable API and may change in future QGIS releases.
+ * \ingroup gui
+ * \since QGIS 3.44
+ */
+class GUI_EXPORT QgsAttributesFormTreeNode
```

Comment:
I'd avoid the use of tree and node throughout. 

We have a model, and we have items, that feels more inline with what we have in QGIS / Qt.

```suggestion
class GUI_EXPORT QgsAttributesFormItem
```